compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -794,8 +793,9 @@ public void visit(BLangMatchStatement matchStatement, AnalyzerData data) {
             analyzeNode(firstClause, data);
         }
         data.failureHandled = failureHandled;
-        analyzeOnFailClause(matchStatement.onFailClause, data);
-        data.errorTypes.pop();
+        if (onFailExists) {
```

Comment:
Do we need this if check here since we check that in `analyzeOnFailClause()`? https://github.com/ballerina-platform/ballerina-lang/blob/27d71e9b649b75036a4ff5fcc6f346f611e3177a/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java#L601

Reply:
Updated in daeb71d752dd2b0ec1b555a64009727b164a51b2